### ISSUE TITLE
Inconsistent Training Graph (Case : Transfer Learning)

### ISSUE THREAD

**Issue by Densus100** [Kamlesh364/Mask-RCNN-TF2.7.0-keras2.7.0#10]
Hello, I'm trying to do transfer learning with mask rcnn. I know that this repo is based on the original mask rcnn by matterport so i tried both of them.

In this case I try to train with total 20 epoch in 2 part for each model. This is the train detail : 
1. 10 epoch layers='heads' (freezing all the backbone layers and training only the randomly initialized layers)
2. 10 epoch layers="all" (fine-tune all layers)

So the **_training log of original matterport mask rcnn_** looks like this : 
the Loss looks very smooth between the 1st part and the 2nd part of training
![alt text](https://i.ibb.co/w4NXF1b/original.png)

==================================================

and this one is **_the training log using this repo_** :
![alt text](https://i.ibb.co/Vp0Txw7/modified.png)

there is a section where the loss value spike significantly unlike the original matterport, and that section is between the 1st part and the 2nd part of training

In Addition there are some problems when I do training with this repo (The result doesn't make sense, the mask is just some random line) so i add this modification in the model.py file as mentioned in here #1 
```
from tensorflow.compat.v1 import ConfigProto
from tensorflow.compat.v1 import InteractiveSession
config = ConfigProto()
config.gpu_options.allow_growth = True
session = InteractiveSession(config=config)
```

I know there are some kind of explanation for loss function in here #7 but i guess this is different from that case.
Comparing both model (this repo vs original matterport) with the same dataset, epoch, and config but ended up with getting 2 different loss result is kinda confusing for me. I hope someone can help me.

Is this some kind of an error or it is supposed to do this way ? 
or is this because of the modification that i add in the model.py ?
Thank you for your time and help. I'm looking forward to any solution or respond  :)

**Comment by Kamlesh364** [comment]
@Densus100 
Please go for more epochs, at-least 100, and then try to compare the final results. 
Do share the comparison results here, as well, to understand the issue better.

regards,
Kamlesh

**Comment by Densus100** [comment]
@Kamlesh364 well I'll try to go for more epochs. 
But I think that's not the actual problem.

The problem is when I tried to switch from transfer learning layers='heads' to layers='all' to fine tune the model no matter how much the epoch is (for now is under 100), the loss is always spikes up. In the other hand when I tried to use the original matterport and do the same thing (heads first then all), the loss is always smooth and no spikes. 

The problem that i want to address is when the training transition happened from transfer learning layers='heads' to layers='all' the loss will always spikes up on the **transition epoch**. (so for example if I train the layers='heads' for 7 epoch then the loss spike will happen in epoch 8 when I train with layers='all')

As your reply in here #11, you suggested that we don't need a lot of epoch for heads right ? So I think if you asked for 100 epoch or more, it doesn't even matter isn't it ? as long as we do the transition from layers='heads' to layers='all' the loss will always spikes up on the transition epoch
for the example as you can see in here #7 and here #11. There are always some loss spikes up significantly when switching from layers='heads' to layers='all'

@Kamlesh364 Do you have any problems when training the model? Please share your results too if possible. Thank you :)

**Comment by Kamlesh364** [comment]
@Densus100 
your issue and Issue #11 seem to be similar. If you find #11 useful, please close this issue and follow the same, otherwise let this issue open.

regrads,
Kamlesh

**Comment by Densus100** [comment]
@Kamlesh364 from the issue #11 the problem is there are a big difference from training loss and val loss. I don't mind the val loss. My concern is this loss graph from #11 also appear a spikes from the transition 5 epoch heads training.

this is the picture that a I took from #11 
![alt text](https://i.ibb.co/YyWvgGP/issue11.png)
Did you see the green circle ?
that's what i mean by loss spikes in the epoch of transition from heads to all

**Comment by Kamlesh364** [comment]
@Densus100
Please confirm if you have tried this for multiple times and getting the same results every time. If this is the case, please try to generate the results for another dataset, and share if still same spike is occurring.

**Comment by Densus100** [comment]
@Kamlesh364 
Yes I've done it multiple times and still get the same result with this Mask RCNN repo. But no problems with the original Matterport Mask RCNN repo. 

and Yes, I've done it with different datasets too, even if it's single class or multiple classes, the problem (loss spikes) is still there.

Here are some more examples which the loss spike occurred:

This is one is 5 head + 5 full using single class
![alt text](https://i.ibb.co/wJjbb3F/5h10f.png)

=====

This one is 10 head + 5 full using single class
![alt text](https://i.ibb.co/Zh1cr9n/10h15f.png)

=====

This one is 10 head + 10 full using multiple class (different dataset)
![alt text](https://i.ibb.co/BGK0dNP/10h10f.png)
